src/Microsoft.Health.Abstractions/Errors/ErrorMessage.cs

```diff
@@ -0,0 +1,50 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Health.Abstractions.Errors
+{
+    public class ErrorMessage
+    {
+        /// <summary>
+        /// A unique identifier for the ErrorMessage
+        /// <remarks>
+        /// This Id can be generated by the caller, or left null.
+        /// If null, the IErrorMessageService will generate an Id.
+        /// </remarks>
+        /// </summary>
+        public string Id { get; set; }
+
+        /// <summary>
+        /// The type of error.
+        /// </summary>
+        public string ErrorType { get; set; }
```

Comment:
Do you foresee each service that writes to the Error Messaging Service will determine its own set of error types, or would it be possible to have a generic set of error types (in which case `ErrorType` can have an enum type with these known values instead)?